### PR TITLE
feat(http): dispatch an dedicated action when 401, 403, 404 http errors

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**
always dispatch an action when a 401, 403, 404 occurs.
create a saga to listen this action and just connect this saga in your saggaMiddleware of your application to manage this errors.

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/README.md [hunk 1]
```diff
@@ -254,8 +259,8 @@ you may change the following using simple props:
...
-- [x] embedable apps
-- [ ] react-router v4
-- [ ] i18n
-- [x] generator
-- [x] actionCreator should become first class
+* [x] embedable apps
+* [ ] react-router v4
+* [ ] i18n
+* [x] generator
+* [x] actionCreator should become first class
```
Comment (jmfrancois) [inline packages/cmf/README.md]: could you move all theses changes on #959 ?

Comment (jare-talend) [reply packages/cmf/README.md]: done

File: packages/cmf/src/middlewares/http/middleware.js [hunk 2]
```diff
@@ -160,6 +168,15 @@ export const httpMiddleware = ({ dispatch }) => next => action => {
 						errorObject.stack.response = response;
 						errorObject.stack.messageObject = JSON.parse(response);
 					} finally {
+						switch (errorObject.stack.status) {
+							case HTTP_STATUS.UNAUTHORIZED:
+							case HTTP_STATUS.FORBIDDEN:
+							case HTTP_STATUS.NOT_FOUND:
+								dispatch(on40xError(errorObject.stack.status));
```
Comment (romainseb) [inline packages/cmf/src/middlewares/http/middleware.js]: Should we dispatch 40x Error now when we have a onError function passed to the httpAction ? 
( @jmfrancois )

File: packages/cmf/src/sagas/http.js [hunk 3]
```diff
@@ -119,6 +119,17 @@ export function* wrapFetch(url, config, method = HTTP_METHODS.GET, payload) {
 	const answer = yield call(httpFetch, url, config, method, payload);
 
 	if (answer instanceof Error) {
+		switch (answer.response.status) {
```
Comment (romainseb) [inline packages/cmf/src/sagas/http.js]: Same question here about how we handle the case of onError passed to the action

**Pull request closed without merging**